pages/product/index.vue

```diff
@@ -336,28 +352,35 @@ export default defineComponent({
       )
 
       if (isValid) {
-        setFulfillment(selectedFulfillmentVal, purchaseLocation.value.code)
+        setFulfillment(selectedFulfillmentVal, shortName, purchaseLocation?.value?.code)
+        checkEligibilityForAddToCart()
       }
     }
 
+    const checkEligibilityForAddToCart = () => {
+      isValidForAddToCart.value =
+        isProductVariationsSelected(product.value) && Boolean(product.value.fulfillmentMethod)
+    }
```

Comment:
Convert `isValidForAddToCart` to a computed property
Remove `checkEligibilityForAddToCart` and its calls

Reply:
done